### PR TITLE
AK: Port BumpAllocator.h to Windows

### PR DESCRIPTION
**Pull request by R-Goc**
Ports BumpAllocator.h to Windows using VirtualAlloc, a Windows API. This is done to avoid additional dependecies for AK.

### CODE REVIEW

**Comment by R-Goc** [comment]
Clang format is wrong here. It tries to switch include order which will be wrong, and not compile.

**Comment by yyny** [comment]
You should probably place those includes in

```
// clang-format off
#include <...>
// clang-format on
```

then.

**Comment by R-Goc** [comment]
This looks like it could be spreading Windows.h around. I'll make it a draft for now.

**Comment by R-Goc** [comment]
So I see 3 possible ways out here:
- 1. Forward declare the functions and macros needed in the header. This is already done in sleep.h As in :
```
#if defined(AK_OS_WINDOWS)
extern "C" __declspec(dllimport) void* __stdcall VirtualAlloc(size_t dwSize, u32 flAllocationType, u32 flProtect);
extern "C" __declspec(dllimport) bool __stdcall VirtualFree(void* lpAddress, size_t dwSize, u32 dwFreeType);
#    define MEM_COMMIT 0x00001000
#    define PAGE_READWRITE 0x04
#    define MEM_RELEASE 0x00008000
#endif
```
- 2 Split into a .cpp file and use winapi VirtualAlloc and VirtualFree, with the same code as currently in the header, this way we don't link mman and don't spread widows.h as it is in a cpp file
- 3 Split into a .cpp file and use mman. This requires linking the mman windows port into AK. 
All of the solutions have the same functionality from what I can see, other than 2 and 3 requiring an additional translation unit.

File: AK/BumpAllocator.h [hunk 1]
```diff
@@ -106,10 +118,19 @@ class BumpAllocator {
 #ifdef AK_OS_SERENITY
                 new_chunk = serenity_mmap(nullptr, m_chunk_size, PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_RANDOMIZED | MAP_PRIVATE, 0, 0, m_chunk_size, "BumpAllocator Chunk");
 #else
+#    if defined(AK_OS_WINDOWS)
```
Comment (gmta) [inline AK/BumpAllocator.h:121]: This can be an `#elif`

**Review by ADKaster — APPROVED** [review]
This is fine for now. In the future we should consider refactoring this to have a .cpp file, as suggested before.

**Pull request merged**